cmds/boot/systemboot/main.go

```diff
@@ -60,6 +155,191 @@ func getSystemFWVersion(si *smbios.Info) (string, error) {
 	return t0.Version, nil
 }
 
+// IpmiOemSetProcessorAndDimmInfo sends the information of proccessors and DIMMs to BMC
+func IpmiOemSetProcessorAndDimmInfo(si *smbios.Info) error {
+	ProcRequestDataParam1 := oemProcProductNameDataParam1{}
+	ProcRequestDataParam2 := oemProcBasicInfoDataParam2{}
+	DIMMRequestDataParam1 := oemDimmLocationDataParam1{}
+	DIMMRequestDataParam2 := oemDimmTypeDataParam2{}
+	DIMMRequestDataParam3 := oemDimmSpeedDatapParam3{}
+	DIMMRequestDataParam4 := oemDimmModulePartNumberDataParam4{}
+	DIMMRequestDataParam5 := oemDimmModuleSerialNumberDataParam5{}
+	DIMMRequestDataParam6 := oemDimmModuleManufacturerIDDataParam6{}
+
+	ipmi, err := ipmi.Open(0)
+	if err != nil {
+		log.Printf("Failed to open ipmi device %v", err)
+		return err
+	}
+	defer ipmi.Close()
+
+	//-----------------------------------------------------------------------------------------
+	// Start - Set Processor Information(0x10)
+	t4, err := si.GetProcessorInfo()
+	if err != nil {
+		return err
+	}
+
+	ProcRequestDataParam1.ManufacturerID[0] = _WIWYNN_ID0
```

Comment:
Isn't there a way to identify the manufacturer and avoid hardcoding it here?

Reply:
I defined a map manufacturer->manufacturer ID, and retrieve the manufacturer name from SMBIOS type 1.